src/tp_worker_thread.c

```diff
@@ -208,15 +207,18 @@ IMPLEMENT_MOCKABLE_FUNCTION(, int, tp_worker_thread_open, TP_WORKER_THREAD_HANDL
             else
             {
                 /*Codes_SRS_TP_WORKER_THREAD_42_041: [ tp_worker_thread_open shall call threadpool_create_work_item with the threadpool, tp_worker_on_threadpool_work and worker_thread. ]*/
-                worker_thread->threadpool_work_item = threadpool_create_work_item(worker_thread->threadpool, tp_worker_on_threadpool_work, worker_thread);
-                if (worker_thread->threadpool_work_item == NULL)
+                THANDLE(THREADPOOL_WORK_ITEM) threadpool_work_item = threadpool_create_work_item(worker_thread->threadpool, tp_worker_on_threadpool_work, worker_thread);
+                if (threadpool_work_item == NULL)
                 {
                     LogError("threadpool_create_work_item failed");
                     /*Codes_SRS_TP_WORKER_THREAD_42_020: [ If there are any errors then tp_worker_thread_open shall fail and return a non-zero value. ]*/
                     result = MU_FAILURE;
                 }
                 else
                 {
+                    /*Codes_SRS_TP_WORKER_THREAD_01_001: [ tp_worker_thread_open shall save the THANDLE(THREADPOOL_WORK_ITEM) for later use by using THANDLE_INITIALIZE_MOVE(THREADPOOL_WORK_ITEM). ]*/
+                    THANDLE_INITIALIZE_MOVE(THREADPOOL_WORK_ITEM)(&worker_thread->threadpool_work_item, &threadpool_work_item);
```

Comment:
>THANDLE_INITIALIZE_MOVE(THREADPOOL_WORK_ITEM)(&worker_thread->threadpool_work_item, &threadpool_work_item); [](http://example.com/codeflow?start=20&length=107)

consider: temporary can be avoided

`THANDLE_INITIALIZE_MOVE(THREADPOOL_WORK_ITEM)(&worker_thread->threadpool_work_item, &(THANDLE(THREADPOOL_WORK_ITEM){threadpool_create_work_item(worker_thread->threadpool, tp_worker_on_threadpool_work, worker_thread)});`

Reply:
I considered it, but I did not like how the UTs expected calls would come on the other side because of weird order of calls now. And the temp will be 100% optimized by the compiler.